### PR TITLE
Chapter on Software Checklist

### PR DESCRIPTION
**Pull request by fdiblen**
### Summary
 Addresses (but does not close) #878. 

### List of changes proposed in this PR (pull-request)

*Adds a new chapter about software checklist*

### What should a reviewer concentrate their feedback on?

- [ ] Everything looks ok?

### Acknowledging contributors
- [ ] The following people should be added to the [table of contributors](https://github.com/alan-turing-institute/the-turing-way/blob/master/README.md#contributors) in the README file: _fdiblen, bouweandela, PabRod_

### CODE REVIEW

File: book/content/software_checklist/software_checklist.md
Comment (PabRod) [inline book/content/software_checklist/software_checklist.md]: Minor suggestion: mention `cff` files in third place

Registering a DOI in Zenodo is easier and, in my opinion, more important than having a `cff` file.

File: book/content/software_checklist/01/conferences_journals_workshops.md
Comment (PabRod) [inline book/content/software_checklist/01/conferences_journals_workshops.md]: Style guidelines: finish bullet points with "."

File: book/content/software_checklist/01/making_software_citable.md
Comment (PabRod) [inline book/content/software_checklist/01/making_software_citable.md]: Suggestion: use proper hyperlink

File: book/content/software_checklist/01/publishing_results.md
Comment (PabRod) [inline book/content/software_checklist/01/publishing_results.md]: Style guidelines: finish bullet points with "."

File: book/content/software_checklist/02/releases.md
Comment (PabRod) [inline book/content/software_checklist/02/releases.md]: Missing "." at the end of sentence

File: book/content/software_checklist/03/checklist_matrix.md
Comment (PabRod) [inline book/content/software_checklist/03/checklist_matrix.md]: Style guidelines: finish bullet points with "."

File: book/content/software_checklist/software_checklist.md
Comment (PabRod) [inline book/content/software_checklist/software_checklist.md]: Style guidelines: finish bullet points with "."

File: book/content/software_checklist/software_checklist.md
Comment (PabRod) [reply book/content/software_checklist/software_checklist.md]: Fixed in commit 0d3c8ee80dbb576866f1498899b0e82f2957aa6a

File: book/content/software_checklist/01/conferences_journals_workshops.md
Comment (PabRod) [reply book/content/software_checklist/01/conferences_journals_workshops.md]: 38703d1093e6ee8242bedd8faba9b17f8532589d fixes this

File: book/content/software_checklist/01/publishing_results.md
Comment (PabRod) [reply book/content/software_checklist/01/publishing_results.md]: 38703d1093e6ee8242bedd8faba9b17f8532589d fixes this

File: book/content/software_checklist/03/checklist_matrix.md
Comment (PabRod) [reply book/content/software_checklist/03/checklist_matrix.md]: 38703d1093e6ee8242bedd8faba9b17f8532589d fixes this

File: book/content/software_checklist/software_checklist.md
Comment (PabRod) [reply book/content/software_checklist/software_checklist.md]: 38703d1093e6ee8242bedd8faba9b17f8532589d fixes this

File: book/content/software_checklist/02/releases.md
Comment (PabRod) [reply book/content/software_checklist/02/releases.md]: 38703d1093e6ee8242bedd8faba9b17f8532589d fixes this

File: book/content/software_checklist/01/making_software_citable.md
Comment (PabRod) [reply book/content/software_checklist/01/making_software_citable.md]: e1f50e706211c37a86013006afe8a757a0ad2fbb fixes this

**Comment by PabRod** [comment]
@fdiblen @mkuzak Is this it? Can we just merge our branch without any confirmation of approval by the Turing Institute?

**Comment by PabRod** [comment]
Related #919

**Comment by mkuzak** [comment]
please add `review request` label and wait for the review

**Comment by mkuzak** [comment]
@fdiblen and @pabrod I have a question about the "e-Science Conferences, Journals, and Workshops". What is the relation to the research software checklist?

**Comment by PabRod** [comment]
I don't know, @mkuzak. But my feeling is that it should be in a different section, especially if we want to keep the _software checklist_ concise.

**Comment by mkuzak** [comment]
ok, so that should not be here than, can you propose other chapter it should go into? Also @malvikasharan maybe you have some suggestions.

File: book/content/software_checklist/01/making_software_citable.md
Comment (mkuzak) [inline book/content/software_checklist/01/making_software_citable.md]:
```suggestion
This has the advantage that it is unambigous and standardized. For journal articles, using
```

File: book/content/software_checklist/01/making_software_citable.md
Comment (mkuzak) [inline book/content/software_checklist/01/making_software_citable.md]:
```suggestion
software, you can get a DOI by publishing it on [Zenodo](https://zenodo.org/) platform:
```

File: book/content/software_checklist/01/making_software_citable.md
Comment (mkuzak) [inline book/content/software_checklist/01/making_software_citable.md]: I don't understand the use of double backticks here, usually names of files are in single backticks in markdown

File: book/content/software_checklist/01/making_software_citable.md
Comment (mkuzak) [inline book/content/software_checklist/01/making_software_citable.md]:
```suggestion
``CITATION.cff`` files in [Software Sustainability Institute blog post
```

File: book/content/software_checklist/01/making_software_citable.md
Comment (mkuzak) [inline book/content/software_checklist/01/making_software_citable.md]:
```suggestion
on citation files](https://www.software.ac.uk/blog/2017-12-12-standard-format-citation-files)).
```

File: book/content/software_checklist/01/making_software_citable.md
Comment (mkuzak) [inline book/content/software_checklist/01/making_software_citable.md]:
```suggestion
Writing ``CITATION.cff`` files by hand is tedious and error-prone. We recommend using [this online application](https://citation-file-format.github.io/cff-initializer-javascript/) and filling in the provided web form.
```

File: book/content/software_checklist/01/making_software_citable.md
Comment (mkuzak) [inline book/content/software_checklist/01/making_software_citable.md]:
```suggestion
including a ``.zenodo.json`` file in the root of your repository. Writing
```

File: book/content/software_checklist/01/making_software_citable.md
Comment (mkuzak) [inline book/content/software_checklist/01/making_software_citable.md]: Those things are covered by the citable-code guide, I don't we should repeat it here, only refer to the guide

File: book/content/software_checklist/01/making_software_citable.md
Comment (mkuzak) [inline book/content/software_checklist/01/making_software_citable.md]: explain how to do it or add link to the description elsewhere

File: book/content/software_checklist/01/publishing_results.md
Comment (mkuzak) [inline book/content/software_checklist/01/publishing_results.md]: I don't understand this section. Is it about handling datasets and results, is it about private / public repos. Moreover, git-lfs is only one of many solutions, and is not really 'publishing' datasets. I would suggest linking to the sections in the book which deals with publishing data.

File: book/content/software_checklist/01/publishing_results.md
Comment (mkuzak) [inline book/content/software_checklist/01/publishing_results.md]: This looks to me like eScience Center specific. It could be generic, but would need to be rephrased.

**Comment by malvikasharan** [comment]
I agree with Mateusz. This chapter does not look like a checklist. Maybe separate the content or add an appropriate header.

Also, can you please pull changes from the master branch and resolve the conflict in the toc.yml with the update file name/path?

**Comment by fdiblen** [comment]
@malvikasharan @mkuzak Thank you for your reviews. Indeed, this PR was not ready to review yet. I basically moved the content from NLeSC guide to here and didn't have enough time to complete the PR. I will complete it soon and ask for your reviews again.

**Comment by mkuzak** [comment]
thank you @fdiblen 
I removed the `request review` label, please put it back when ready.

**Comment by PabRod** [comment]
By the way, the `review request` label was set by me.

Sorry for that!

**Comment by mkuzak** [comment]
that's ok @PabRod we did review it :)
now it can be improved and the label can be added again when ready

File: book/content/software_checklist/01/making_software_citable.md
Comment (fdiblen) [reply book/content/software_checklist/01/making_software_citable.md]: this line was removed

File: book/content/software_checklist/01/making_software_citable.md
Comment (fdiblen) [reply book/content/software_checklist/01/making_software_citable.md]: chapter was removed

File: book/content/software_checklist/01/making_software_citable.md
Comment (fdiblen) [reply book/content/software_checklist/01/making_software_citable.md]: section was removed

File: book/content/software_checklist/01/making_software_citable.md
Comment (fdiblen) [reply book/content/software_checklist/01/making_software_citable.md]: section was removed

File: book/content/software_checklist/01/publishing_results.md
Comment (fdiblen) [reply book/content/software_checklist/01/publishing_results.md]: section was removed

File: book/content/software_checklist/01/publishing_results.md
Comment (fdiblen) [reply book/content/software_checklist/01/publishing_results.md]: section was removed

File: book/content/software_checklist/01/checklist_matrix.md
Comment (mkuzak) [inline book/content/software_checklist/01/checklist_matrix.md:1]: I don't know what WIP means. Is that Work In Progress? If so, this should not be in the content.
There is no word 'checkmatrix', maybe use 'checklist matrix' instead?

File: book/content/software_checklist/01/checklist_matrix.md
Comment (mkuzak) [inline book/content/software_checklist/01/checklist_matrix.md:3]:
```suggestion
This matrix shows which parts of the software sustainability checklist should be taken care of at (perhaps slightly before) which stage of a project.
```

File: book/content/software_checklist/01/checklist_matrix.md
Comment (mkuzak) [inline book/content/software_checklist/01/checklist_matrix.md:12]:
```suggestion
- [*Pre-release phase*](checklist/checklist_prerelease.md). Eventually, you get the software you intend to keep. Signs you could be in this phase:
```

File: book/content/software_checklist/01/checklist_matrix.md
Comment (mkuzak) [inline book/content/software_checklist/01/checklist_matrix.md:17]:
```suggestion
- [*Maturity phase*](checklist/checklist_mature.md): Software that has reached maturity has a clear function and scope and is used. Signs you could be in this phase:
```

File: book/content/software_checklist/01/checklist_matrix.md
Comment (mkuzak) [inline book/content/software_checklist/01/checklist_matrix.md:19]:
```suggestion
  - The software has users: people using your software/code.
```

File: book/content/software_checklist/01/checklist_matrix.md
Comment (mkuzak) [inline book/content/software_checklist/01/checklist_matrix.md:20]:
```suggestion
  - You have an external contributor.
```

File: book/content/software_checklist/software_checklist.md
Comment (mkuzak) [inline book/content/software_checklist/software_checklist.md:40]: 1. the contrition guidelines are above
2. why this link leads to code of conduct?

**Comment by fdiblen** [comment]
@malvikasharan @mkuzak Sorry for the delay. I cannot add a new label but this PR is ready to review.

File: book/content/software_checklist/software_checklist.md
Comment (fdiblen) [reply book/content/software_checklist/software_checklist.md]: done

File: book/content/software_checklist/01/checklist_matrix.md
Comment (fdiblen) [reply book/content/software_checklist/01/checklist_matrix.md]: done

File: book/content/software_checklist/01/checklist_matrix.md
Comment (fdiblen) [reply book/content/software_checklist/01/checklist_matrix.md]: done

File: book/content/software_checklist/01/checklist_matrix.md
Comment (fdiblen) [reply book/content/software_checklist/01/checklist_matrix.md]: done

File: book/content/software_checklist/01/checklist_matrix.md
Comment (fdiblen) [reply book/content/software_checklist/01/checklist_matrix.md]: done

File: book/content/software_checklist/01/checklist_matrix.md
Comment (fdiblen) [reply book/content/software_checklist/01/checklist_matrix.md]: done

File: book/content/software_checklist/01/checklist_matrix.md
Comment (fdiblen) [reply book/content/software_checklist/01/checklist_matrix.md]: done

**Comment by mkuzak** [comment]
thank you @fdiblen I think that is ok now.
@malvikasharan can you let us know it this is ready?

**Review by malvikasharan — APPROVED** [review]

**Comment by c-martinez** [comment]
@malvikasharan , @fdiblen -- this PR was never merged, and seems to now conflict with master. Also, as pointed by @bouweandela, checklists are now included after every chapter.

How to proceed? Should we close this PR and contribute to the existing checklists instead?

**Comment by fdiblen** [comment]
> @malvikasharan , @fdiblen -- this PR was never merged, and seems to now conflict with master. Also, as pointed by @bouweandela, checklists are now included after every chapter.
> 
> How to proceed? Should we close this PR and contribute to the existing checklists instead?

@c-martinez It is fine with me if you want to close this PR and keep checklist in each chapter.

**Comment by c-martinez** [comment]
@malvikasharan agrees, so lets close it and keep the checklists which are already there.

**Pull request closed without merging**
